source/core/shape.cc

```diff
@@ -980,43 +950,67 @@ CompositeShape<dim>::value_with_cell_guess(
   Point<dim> centered_point = this->align_and_center(evaluation_point);
 
   // The levelset value of all component shapes is computed
-  std::map<unsigned int, double> components_value;
+  std::map<unsigned int, double>         components_value;
+  std::map<unsigned int, Tensor<1, dim>> dummy_components_gradient;
   for (auto const &[component_id, component] : constituents)
     {
       components_value[component_id] =
         component->value_with_cell_guess(centered_point, cell);
+      dummy_components_gradient[component_id] = Tensor<1, dim>{};
     }
 
-  // The boolean operations between the shapes are applied in order
-  // The last computed levelset value is considered to be the right value
-  double levelset = components_value[0];
-  for (auto const &[operation_id, op_triplet] : operations)
+  double levelset;
+  std::tie(levelset, std::ignore) =
+    apply_boolean_operations(components_value, dummy_components_gradient);
```

Comment:
The name of the variables you use to apply boolean operations here is very different from the one in the .h, Can you document it a bit more? What is dummy_components_gradients? Is it the gradients for each shape? Then use a more verbose name that describes that it is.